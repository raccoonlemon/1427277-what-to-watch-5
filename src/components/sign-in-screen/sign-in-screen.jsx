import PropTypes from 'prop-types';
import React, {createRef, PureComponent} from "react";
import {connect} from "react-redux";
import {login} from "../../store/api-actions";
import Footer from "../footer/footer";
import Header from "../header/header";

export class SignInScreen extends PureComponent {

  constructor(props) {
    super(props);

    this.loginRef = createRef();
    this.passwordRef = createRef();

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    const {onSubmit} = this.props;
    evt.preventDefault();

    onSubmit({
      login: this.loginRef.current.value,
      password: this.passwordRef.current.value,
    });
  }

  render() {
    return (
      <div className="user-page">
        <Header className="user-page__head" needToShowUserMenu={false}>
          <h1 className="page-title user-page__title">Sign in</h1>
        </Header>

        <div className="sign-in user-page__content">
          <form action="#" className="sign-in__form">
            <div className="sign-in__fields">
              <div className="sign-in__field">
                <input
                  ref={this.loginRef}
                  className="sign-in__input"
                  type="email"
                  placeholder="Email address"
                  name="user-email"
                  id="user-email" />
                <label className="sign-in__label visually-hidden" htmlFor="user-email">Email address</label>
              </div>
              <div className="sign-in__field">
                <input
                  ref={this.passwordRef}
                  className="sign-in__input"
                  type="password"
                  placeholder="Password"
                  name="user-password"
                  id="user-password" />
                <label className="sign-in__label visually-hidden" htmlFor="user-password">Password</label>
              </div>
            </div>
            <div className="sign-in__submit">
              <button
                className="sign-in__btn"
                type="submit"
                onClick={this.handleSubmit}>Sign in</button>
            </div>
          </form>
        </div>

        <Footer/>
      </div>);
  }
}

SignInScreen.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit(authData) {
    dispatch(login(authData));
  }
});

export default connect(null, mapDispatchToProps)(SignInScreen);
