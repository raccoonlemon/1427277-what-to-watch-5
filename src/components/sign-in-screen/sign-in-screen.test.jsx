import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {SignInScreen} from "./sign-in-screen";

const renderer = new ShallowRenderer();
const noop = ()=>{};

it(`<SignInScreen> renders correctly`, () => {
  renderer.render(
      <SignInScreen
        isRequested={false}
        isRequestFailed = {false}
        errorCode = {0}
        onSubmitAction = {noop}
      />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
