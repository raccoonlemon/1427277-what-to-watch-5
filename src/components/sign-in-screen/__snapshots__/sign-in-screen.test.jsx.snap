// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInScreen> renders correctly 1`] = `
<div
  className="user-page"
>
  <Header
    className="user-page__head"
    needToShowUserMenu={false}
  >
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
  </Header>
  <div
    className="sign-in user-page__content"
  >
    <form
      action="#"
      className="sign-in__form"
    >
      <div
        className="sign-in__message"
      />
      <div
        className="sign-in__fields"
      >
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-email"
            name="user-email"
            onChange={[Function]}
            placeholder="Email address"
            type="email"
            value=""
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-email"
          >
            Email address
          </label>
        </div>
        <div
          className="sign-in__field"
        >
          <input
            className="sign-in__input"
            id="user-password"
            name="user-password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <label
            className="sign-in__label visually-hidden"
            htmlFor="user-password"
          >
            Password
          </label>
        </div>
      </div>
      <div
        className="sign-in__submit"
      />
    </form>
  </div>
  <Footer
    className=""
  />
</div>
`;
